src/__tests__/git.test.ts

```diff
@@ -0,0 +1,86 @@
+import {mkdirSync, rmdirSync, existsSync} from 'fs'
+import * as git from '../git'
+import * as spawn from '../spawn'
+import * as core from '@actions/core'
+
+describe('git', () => {
+  // We don't want to _actually_ spawn external commands, so we mock the function
+  let spawnSpy: jest.SpyInstance
+  // Capture the startGroup calls
+  let coreSpy: jest.SpyInstance
+
+  beforeEach(() => {
+    coreSpy = jest.spyOn(core, 'startGroup')
+    spawnSpy = jest.spyOn(spawn, 'spawn').mockResolvedValue({
+      // 0 is the exit code for success
+      exitCode: 0
+    })
+    // We don't want to _actually_ clone the repo, so we mock the function
+    jest.spyOn(git, 'clone').mockResolvedValue()
+
+    // The script removes the .tmp folder at the end, so let's create it
```

Comment:
Can we mock `mkdirSync()`/`rmdirSync()` instead and verify that they were called?